[EG] System events reformat

l0lawrence opened the pull request.

l0lawrence committed:
refactor

l0lawrence committed:
updayrd

l0lawrence committed:
format

l0lawrence committed:
add apicenter

l0lawrence committed:
update new events

l0lawrence committed:
tsp format

l0lawrence committed:
new events

l0lawrence committed:
modelAsString for Media Union as False

l0lawrence committed:
tsp format

l0lawrence committed:
Revert "tsp format"

This reverts commit 1a01cf8de33e48e80ccde56cf1748f68a4ff86b8.

l0lawrence committed:
Revert "modelAsString for Media Union as False"

This reverts commit 6b605812192def3ba6baef1894f53ea5ea71c854.

l0lawrence committed:
remove ReadOnly

l0lawrence committed:
rename resources

l0lawrence committed:
remove duplicate vault

l0lawrence committed:
remove Optional from Nullable Types part 1

l0lawrence committed:
add csharp to x-ms-client-names

l0lawrence committed:
typo

l0lawrence committed:
swagger typo

l0lawrence committed:
formatting

richardpark-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Media/MediaServices.tsp:
```diff
@@ -153,7 +144,7 @@ namespace Microsoft.EventGrid.SystemEvents {
     `@odataType`: string;
```
I believe the `@discriminator` field here should be `@odata.type` (note the added '.'):

Based on the events they have listed here: 

https://learn.microsoft.com/en-us/azure/event-grid/event-schema-media-services?tabs=cloud-event-schema#joboutputstatechange

```json
   "@odata.type": "#Microsoft.Media.JobOutputAsset",
```

richardpark-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.EventGrid/EventGrid.tsp:
```diff
@@ -55,13 +55,11 @@ model SubscriptionValidationEventData {
   /** The validation code sent by Azure Event Grid to validate an event subscription.
    * To complete the validation handshake, the subscriber must either respond with this validation code as part of the validation response,
    * or perform a GET request on the validationUrl (available starting version 2018-05-01-preview). */
```
(apologies, can't leave a comment on unchanged code)

A few of the comments in here make reference to IDs that aren't valid (they have `SystemEvents` in them, but it's not in their online docs):

* "Microsoft.EventGrid.SystemEvents.MQTTClientCreatedOrUpdated"
* "Microsoft.EventGrid.SystemEvents.MQTTClientDeleted"
* "Microsoft.EventGrid.SystemEvents.MQTTClientSessionConnected",
* "Microsoft.EventGrid.SystemEvents.MQTTClientSessionDisconnected"
* "Microsoft.EventGrid.SystemEvents.SubscriptionDeletedEvent",
* "Microsoft.EventGrid.SystemEvents.SubscriptionValidationEvent"

The MQTT ones are here, for example:
https://learn.microsoft.com/en-us/azure/event-grid/event-schema-event-grid-namespace?tabs=cloud-event-schema

I just noticed this when I regen'd and they came up differently than last time.

richardpark-msft replied:
I ended up adding these to my PR to this PR. So SOON.

richardpark-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -692,78 +692,78 @@ model AcsRouterWorkerSelector {
   /** Router Job Worker Selector Time to Live in Seconds */
   @encode("ISO8601") // TODO verify this is timespan in C#
-  ttlSeconds?: duration;
+  ttlSeconds: duration;
```
There is something odd going on here. I have an older version of the JSON for this that lists 'ttlSeconds' as a float, not a string (or duration).

Looking at the swagger I see it's the case there as well: https://github.com/Azure/azure-rest-api-specs/blob/44cbf51b9eee8a2dbf55ced61a61032aaed0148b/specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json#L2167

HOWEVER, looking at the user-facing docs, they've doc'd it as a string! So I'm not 100% on what is right here - did they make a breaking change at some point?

https://learn.microsoft.com/en-us/azure/event-grid/communication-services-router-events#microsoftcommunicationrouterjobqueued

CC: @lmazuel

richardpark-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -692,78 +692,78 @@ model AcsRouterWorkerSelector {
 
   /** Router Job Worker Selector Label Operator */
-  operator?: AcsRouterLabelOperator;
+  operator: AcsRouterLabelOperator;
```
Is this actually supposed to deserialize from 'labelOperator', not 'operator'?

(https://learn.microsoft.com/en-us/azure/event-grid/communication-services-router-events#microsoftcommunicationrouterjobqueued)

richardpark-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -692,78 +692,78 @@ model AcsRouterWorkerSelector {
 
   /** Router Job Worker Selector State */
-  selectorState?: AcsRouterWorkerSelectorState;
+  selectorState: AcsRouterWorkerSelectorState;
```
Looks like this is actually under 'state', not 'selectorState':

https://github.com/Azure/azure-rest-api-specs/blob/44cbf51b9eee8a2dbf55ced61a61032aaed0148b/specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json#L2171

sarangan12 committed:
Client Name Overrides JS EG System Events

Commit:
Adding in renames for Go

Commit:
fix RecordingDurationMs casing

Commit:
Fixing something that Libba refused to for like months now.

Commit:
Removing SystemEvents from the constant names

Commit:
Should be a number == float64.

Commit:
Field name is labelOperator, not operator.

(as shown here: https://github.com/Azure/azure-rest-api-specs/blob/0a8300f818c1df4018e2586540bed4c72c7c5375/specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json#L2120)

Commit:
Field is state, not selectorState:

https://github.com/Azure/azure-rest-api-specs/blob/0a8300f818c1df4018e2586540bed4c72c7c5375/specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json#L2171

Commit:
More renames to bring us to par with what we had before.

Commit:
Adding in some docs that were missing for unions.

l0lawrence committed:
format

l0lawrence committed:
fix warnigns

l0lawrence committed:
moving around clientNames

l0lawrence committed:
spellcheck

l0lawrence committed:
Merge branch 'main' into system_events_format

l0lawrence committed:
Merge branch 'main' into system_events_format

l0lawrence committed:
Merge branch 'main' into system_events_format

l0lawrence committed:
try to update csharp clientNames

l0lawrence committed:
tsp format

l0lawrence committed:
missing labelValue x-ms-client-name

l0lawrence committed:
format

l0lawrence committed:
nit typo

l0lawrence committed:
update csharp
